ffi/derive/src/convert.rs

```diff
@@ -20,140 +27,182 @@ pub fn derive_ffi_type(mut input: DeriveInput) -> TokenStream {
         Data::Enum(item) => {
             let repr = find_attr(&input.attrs, "repr");
 
-            if item.variants.is_empty() {
-                abort!(name, "uninhabited enum's cannot be instantiated");
-            }
-
             if is_fieldless_enum(&item) {
+                if item.variants.len() == 1 {
+                    // NOTE: one-variant fieldless enums have representation of ()
+                    return derive_ffi_type_for_opaque_item(name, &input.generics);
+                }
+                if without_repr(&input.attrs) {
+                    abort!(name, "Fieldless enum missing #[repr(int)]");
+                }
                 if item.variants.iter().any(|v| v.discriminant.is_some()) {
                     abort!(
                         name,
-                        "fieldless enums with explicit discriminants are prohibited"
+                        "Fieldless enums with explicit discriminants are prohibited"
```

Comment:
I can't leave this diff comment for a proper line, but here it is:

```rust
if let Some(variant) = item.variants.iter().find(|v| v.discriminant.is_some()) {
    abort!(
        variant,
        "Fieldless enums with explicit discriminants are prohibited",
    )
}
```

I think with that it will show an error for a concrete variant